bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/api/AzureActions.java

```diff
@@ -0,0 +1,147 @@
+        this(httpClientFactory, accountName, containerName, "");
+    }
+
+    public AzureActions(HttpClientFactory httpClientFactory, String accountName, String containerName,
+            String azureAccessKey) {
+        this.httpClient = httpClientFactory.getCommonHttpClient();
+        try {
+            this.containerUri = new URL("https://" + accountName + ".blob.core.windows.net/" + containerName);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("Unable to parse service endpoint: " + e.getMessage());
+        }
+        this.azureAccessKey = azureAccessKey;
+        this.accountName = accountName;
+        this.containerName = containerName;
+    }
+
+    public List<String> listContainer(String prefix) throws Exception {
+        Map<String, String> headers = new HashMap<String, String>();
+        Map<String, String> params = new HashMap<String, String>();
+        return listBlob(prefix, headers, params);
+    }
+
+    public List<String> listBlob(String prefix, Map<String, String> headers, Map<String, String> params)
+            throws Exception {
+
+        params.put("restype", "container");
+        params.put("comp", "list");
+        params.put("maxresults", "1000");
+        params.put("prefix", prefix);
+        headers.put(ACCEPT.toString(), CONTENT_TYPE);
+
+        if (!azureAccessKey.isEmpty()) {
+            Azure4SignerForAuthorizationHeader signer = new Azure4SignerForAuthorizationHeader("GET",
+                    this.containerUri);
+            String authorization;
+            try {
+                authorization = signer.computeSignature(headers, params, accountName, azureAccessKey, containerName);
+            } catch (HttpUtilException e) {
+                throw new AuthException(e);
+            }
+            headers.put("Authorization", authorization);
+        }
+
+        Request request = httpClient.newRequest(this.containerUri.toString()) //
+                .method(GET) //
+                .timeout(REQUEST_TIMEOUT.toNanos(), TimeUnit.NANOSECONDS); //
+
+        for (String headerKey : headers.keySet()) {
+            request.header(headerKey, headers.get(headerKey));
+        }
+        for (String paramKey : params.keySet()) {
+            request.param(paramKey, params.get(paramKey));
+        }
+
+        ContentResponse contentResponse = request.send();
+        if (contentResponse.getStatus() != 200) {
+            throw new Exception("HTTP Response is not 200");
+        }
+
+        DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder docBuilder = docBuilderFactory.newDocumentBuilder();
+        // This returns extra character before <xml. Need to find out why
+        String sResponse = contentResponse.getContentAsString();
+        InputSource is = new InputSource(new StringReader(sResponse.substring(sResponse.indexOf("<"))));
```

Comment:
It could be made a little bit more robust when eeusing a regex and search for <?xml instead of the < that can be part of any response. Just need to prevend exceptions here.